### PR TITLE
Endpoints for post comments

### PR DESCRIPTION
**Pull request by Devorein**

### CODE REVIEW

File: pages/api/forums/posts/[postId]/comments/[commentId].ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+const handler = nc<NextApiRequest, NextApiResponse>({ onError, onNoMatch });
+
+handler.use(requireUser).put(updatePostCommentHandler).delete(deletePostCommentHandler);
+
+async function updatePostCommentHandler(req: NextApiRequest, res: NextApiResponse<PageComment>) {
+  const { postId, commentId } = req.query as any as { postId: string; commentId: string };
+  const body = req.body as UpdatePostCommentInput;
+  const userId = req.session.user.id;
+
+  const post = await prisma.post.findUnique({
```
Comment (msadura) [inline pages/api/forums/posts/[postId]/comments/[commentId].ts:23]: please refrain from using db queries in handlers. Instead the should be a helper "getPostById" helper in `lib`

File: pages/api/forums/posts/[postId]/comments/[commentId].ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  const postComment = await updatePostComment({ commentId, userId, ...body });
+
+  res.status(200).json(postComment);
+}
+
+async function deletePostCommentHandler(req: NextApiRequest, res: NextApiResponse) {
+  const { postId, commentId } = req.query as any as { postId: string; commentId: string };
+  const userId = req.session.user.id;
+
+  const post = await prisma.post.findUnique({
```
Comment (msadura) [inline pages/api/forums/posts/[postId]/comments/[commentId].ts:59]: same as in update. Helper > code copy paste.

Comment (msadura) [reply pages/api/forums/posts/[postId]/comments/[commentId].ts:59]: actually whole flow of checking if post exists + access check can be moved to single utility function

File: pages/api/forums/posts/[postId]/comments/index.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+  res.status(200).json(postCommentsWithVotes);
+}
+
+async function createPostCommentHandler(req: NextApiRequest, res: NextApiResponse<PostCommentWithVote>) {
+  const { postId } = req.query as any as { postId: string };
+  const body = req.body as CreatePostCommentInput;
+  const userId = req.session.user.id;
+
+  const post = await prisma.post.findUnique({
```
Comment (msadura) [inline pages/api/forums/posts/[postId]/comments/index.ts:32]: and the same as previously ;)

File: testing/setupDatabase.ts [hunk 4]
```diff
@@ -472,6 +475,60 @@ export function createPage(
   }) as Promise<IPageWithPermissions>;
 }
 
+export async function generatePostComment({ userId, spaceId }: { spaceId: string; userId: string }) {
```
Comment (msadura) [inline testing/setupDatabase.ts:478]: please move those functions to `testing/utils/forum` (or `/forums` or `/posts`). This file is huge already and it is time to organize testing helpers a bit better.

**Review by msadura — CHANGES_REQUESTED** [review]

File: testing/setupDatabase.ts (hunk 4 above)
Comment (motechFR) [reply testing/setupDatabase.ts:478]: Agreed

File: lib/forums/comments/__tests__/updatePostComment.spec.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+let user: User;
+
+beforeAll(async () => {
+  const generated = await generateUserAndSpaceWithApiToken();
+  space = generated.space;
+  user = generated.user;
+});
+
+describe('updatePostComment', () => {
+  it('should update a post comment if the user is the creator', async () => {
```
Comment (motechFR) [inline lib/forums/comments/__tests__/updatePostComment.spec.ts:20]: @Devorein For lib methods, we 99% of times don't have access controls. These are enforced on API layer (see the forum / categories libs + APIs for this).

This is a case where we want to respect this pattern. Can you please move access controls up to the lib and associated tests there?

For update method libs, you mainly want to test
- The keys the update method can update (as some keys might be only updated via other lib actions such as lock / unlock bounty)
- The return value of the update method

**Review by motechFR — CHANGES_REQUESTED** [review]
Please move the lib access controls to the API, as well as associated tests

File: pages/api/forums/posts/[postId]/comments/[commentId].ts (hunk 1 above)
Comment (Devorein) [reply pages/api/forums/posts/[postId]/comments/[commentId].ts:23]: Updated thnx. There was already a `lib` helper called `getForumPost`

**Pull request merged**